From 606483decf1701d23a61fbd80f024984bc5e390a Mon Sep 17 00:00:00 2001
From: Mason Chang <mason-cw.chang@mediatek.com>
Date: Tue, 23 Sep 2025 17:03:40 +0800
Subject: [PATCH] arm64: dts: mediatek: mt7988: add multiple thermal zones

Expose the additional MT7988 LVTS sensors as separate thermal zones so the
SoC reports CPU, 2.5G Ethernet, TOPS and ETHWARP temperature domains.

Source: MTK OpenWrt 25.12
  target/linux/mediatek/patches-6.12/
  999-dts-02-arm64-dts-mediatek-add-mt7988-multiple-thermal-zones.patch

Adaptation notes:
  - Port the DTS-only change from Linux 6.12 to the Linux 6.18 source layout.
  - Keep the thermal binding include in this patch so the sensor constants do
    not depend on an unrelated CPU frequency patch.
  - Do not change LVTS driver behavior or add cooling policy.

Signed-off-by: Mason Chang <mason-cw.chang@mediatek.com>
Signed-off-by: Codex <codex@openai.com>
---
 arch/arm64/boot/dts/mediatek/mt7988a.dtsi | 94 ++++++++++++++++++++++-
 1 file changed, 93 insertions(+), 1 deletion(-)

--- a/arch/arm64/boot/dts/mediatek/mt7988a.dtsi
+++ b/arch/arm64/boot/dts/mediatek/mt7988a.dtsi
@@ -5,6 +5,7 @@
 #include <dt-bindings/phy/phy.h>
 #include <dt-bindings/pinctrl/mt65xx.h>
 #include <dt-bindings/reset/mediatek,mt7988-resets.h>
+#include <dt-bindings/thermal/mediatek,lvts-thermal.h>
 
 / {
 	compatible = "mediatek,mt7988a";
@@ -1012,12 +1013,103 @@
 		cpu_thermal: cpu-thermal {
 			polling-delay-passive = <1000>;
 			polling-delay = <1000>;
-			thermal-sensors = <&lvts 0>;
+			thermal-sensors = <&lvts MT7988_CPU_0>;
 			trips {
 				cpu_trip_crit: crit {
 					temperature = <125000>;
 					hysteresis = <2000>;
 					type = "critical";
+				};
+			};
+		};
+
+		cpu_1_thermal: cpu-1-thermal {
+			polling-delay-passive = <1000>;
+			polling-delay = <1000>;
+			thermal-sensors = <&lvts MT7988_CPU_1>;
+			trips {
+				cpu_1_trip_crit: crit {
+					temperature = <125000>;
+					hysteresis = <2000>;
+					type = "critical";
+				};
+			};
+		};
+
+		eth2p5g_thermal: eth2p5g-thermal {
+			polling-delay-passive = <1000>;
+			polling-delay = <1000>;
+			thermal-sensors = <&lvts MT7988_ETH2P5G_0>;
+			trips {
+				eth2p5g_trip_crit: crit {
+					temperature = <125000>;
+					hysteresis = <2000>;
+					type = "critical";
+				};
+			};
+		};
+
+		eth2p5g_1_thermal: eth2p5g-1-thermal {
+			polling-delay-passive = <1000>;
+			polling-delay = <1000>;
+			thermal-sensors = <&lvts MT7988_ETH2P5G_1>;
+			trips {
+				eth2p5g_1_trip_crit: crit {
+					temperature = <125000>;
+					hysteresis = <2000>;
+					type = "critical";
+				};
+			};
+		};
+
+		tops_thermal: tops-thermal {
+			polling-delay-passive = <1000>;
+			polling-delay = <1000>;
+			thermal-sensors = <&lvts MT7988_TOPS_0>;
+			trips {
+				tops_trip_crit: crit {
+					temperature = <125000>;
+					hysteresis = <2000>;
+					type = "critical";
+				};
+			};
+		};
+
+		tops_1_thermal: tops-1-thermal {
+			polling-delay-passive = <1000>;
+			polling-delay = <1000>;
+			thermal-sensors = <&lvts MT7988_TOPS_1>;
+			trips {
+				tops_1_trip_crit: crit {
+					temperature = <125000>;
+					hysteresis = <2000>;
+					type = "critical";
+				};
+			};
+		};
+
+		ethwarp_thermal: ethwarp-thermal {
+			polling-delay-passive = <1000>;
+			polling-delay = <1000>;
+			thermal-sensors = <&lvts MT7988_ETHWARP_0>;
+			trips {
+				ethwarp_trip_crit: crit {
+					temperature = <125000>;
+					hysteresis = <2000>;
+					type = "critical";
+				};
+			};
+		};
+
+		ethwarp_1_thermal: ethwarp-1-thermal {
+			polling-delay-passive = <1000>;
+			polling-delay = <1000>;
+			thermal-sensors = <&lvts MT7988_ETHWARP_1>;
+			trips {
+				ethwarp_1_trip_crit: crit {
+					temperature = <125000>;
+					hysteresis = <2000>;
+					type = "critical";
 				};
 			};
 		};
